lib/basedriver/device-settings.js

```diff
@@ -3,7 +3,30 @@ import log from './logger';
 import { DEFAULT_MATCH_THRESHOLD } from './commands/images';
 
 const GLOBAL_DEFAULT_SETTINGS = {
+  // value between 0 and 1 representing match strength, below which an image
+  // element will not be found
   imageMatchThreshold: DEFAULT_MATCH_THRESHOLD,
+
+  // if the image returned by getScreenshot differs in size or aspect ratio
+  // from the screen, attempt to fix it automatically
+  fixImageFindScreenshotDims: true,
+
+  // whether Appium should ensure that an image template sent in during image
+  // element find should have its size adjusted so the match algorithm will not
+  // complain
+  fixImageTemplateSize: false,
+
+  // whether Appium should re-check that an image element can be matched
+  // against the current screenshot before clicking it
+  checkForImageElementStaleness: true,
+
+  // whether before clicking on an image element Appium should re-determine the
+  // position of the element on screen
+  autoUpdateImageElementPosition: false,
+
+  // which method to use for tapping by coordinate for image elements. the
+  // options are 'w3c' or 'mjsonwp'
+  imageElementTapStrategy: 'w3c',
```

Comment:
shouldn't this be detected automatically?

Reply:
how can it be detected automatically? we don't know which functionality the downstream driver supports. i.e., this doesn't (and shouldn't) correspond to the protocol used. it corresponds to the functionality available in the driver.

Reply:
we know in which mode the downstream driver is running (w3c/jsonwp), don't we?

Reply:
I'm not sure that we do. And anyway the mode doesn't matter: all we're doing is calling `driver.performActions` or `driver.performTouch`. There's no protocol involved. I'm just giving access to these methods protocol-related names because that's how they're known. But maybe strings like `w3cAction` and `touchAction` would be better.

But your comment makes me realize that this will not be proxied so in the case of WDA, we'll have to ensure the xcuitest driver has `performActions` on it, and manually proxy the command forward to WDA. That's not ideal.